backend/stylists/app/Http/Controllers/TipController.php

```diff
@@ -0,0 +1,237 @@
+        
+        $view_properties['search'] = $request->input('search');
+        $view_properties['exact_word'] = $request->input('exact_word');
+        $view_properties['from_date'] = $request->input('from_date');
+        $view_properties['to_date'] = $request->input('to_date');
+
+        $view_properties['min_price'] = $request->input('min_price');
+        $view_properties['max_price'] = $request->input('max_price');
+        
+        $view_properties['app_sections'] = AppSections::all();
+        $view_properties['logged_in_stylist_id'] = $user_data->id;
+        $view_properties['popup_entity_type_ids'] = $entity_nav_tabs;
+        $view_properties['entity_type_to_send'] = EntityType::LOOK;
+        $view_properties['recommendation_type_id'] = RecommendationType::MANUAL;
+        $view_properties['is_owner_or_admin'] = Auth::user()->hasRole('admin');
+        
+        
+        return view('tip.list', $view_properties);
+
+    }
+  
+    public function getView()
+    {
+        $tip                = Tip::find($this->resource_id);
+        $view_properties    = [];
+        
+        if ($tip) {
+            
+            $products           = $tip->products;
+            $status             = Status::find($tip->status_id);
+            
+            $view_properties    = array('tip' => $tip, 'products' => $products, 'stylist' => $tip->stylist, 'status' => $status);
+            $view_properties['is_owner_or_admin'] = Auth::user()->hasRole('admin') || $tip->stylist_id == Auth::user()->id;
+            
+            
+        } else {
+            
+            return view('404', array('title' => 'Tip not found'));
+        }
+        
+        return view('tip.view', $view_properties);
+    }
+    
+    public function getEdit($id)
+    {
+        $tip = Tip::find($this->resource_id);
+        $view_properties = null;
+        
+        if ($tip) {
+            
+            $lookup = new Lookup();
+            
+            $view_properties['tip']             = $tip;
+            $view_properties['gender_id']       = intval($tip->gender_id);
+            $view_properties['genders']         = $lookup->type('gender')->get();
+            $view_properties['status_id']       = intval($tip->status_id);
+            $view_properties['statuses']        = $lookup->type('status')->get();
+            $view_properties['occasion_id']     = intval($tip->occasion_id);
+            $view_properties['occasions']       = $lookup->type('occasion')->get();
+            $view_properties['age_group_id']    = intval($tip->age_group_id);
+            $view_properties['age_groups']      = $lookup->type('age_group')->get();
+            $view_properties['budget_id']       = intval($tip->budget_id);
+            $view_properties['budgets']         = $lookup->type('budget')->get();
+            $view_properties['body_type_id']    = intval($tip->body_type_id);
+            $view_properties['body_types']      = $lookup->type('body_type')->get();
+            
```

Comment:
pass only tip to view and find genders, statuses, occasions, age_groups, budgets, body_types as tips relation using with() query.